railties/test/application/test_runner_test.rb

```diff
@@ -14,7 +15,7 @@ def teardown
       teardown_app
     end
 
-    def test_run_in_test_environment
+    def test_run_in_differing_environments
```

Comment:
I must admit the setup line length got me to consolidate the tests. I do think you're right about keeping the old method, so we get a better test name, :+1: